### PR TITLE
[Feature] Schedule appointment within a specific date range

### PR DESCRIPTION
**Pull request by yair-dana**
## Description
This PR adds a new feature to the system which allows users to schedule appointments within a specific date range.
1) The default start date is set to today's date, but users can now choose their own start date. 
2) The date format has been fixed to conform to the Israeli format of `DD-MM-YYYY`.

### Open Question
There is a decision to be made on how to implement this feature - either by calling the MyVisit service with the updated Start Date, or by filtering the dates after they have been fetched from the service. (For now- both approaches are implemented)
My concern is that the `startDate` functionality may not be properly implemented on the MyVisit side.
WDYT?

## Type of change

Please delete options that are not relevant.

-   [x] New feature (non-breaking change which adds functionality)
-   [x] This change requires a documentation update

## How Has This Been Tested?

This PR adds more unit tests and an RTL driver to improve the testing capabilities of the BOT.
The unit tests check the Date functionality (`isBefore`/`isAfter`/`isDateInRange`), while the RTL driver and multiple tests cover the `DateRangePicker` component. 
These additions will increase the reliability of the system, ensuring a better user experience and reducing the risk of issues in the future.

## Checklist:

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [x] My changes generate no new warnings
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] New and existing unit tests pass locally with my changes
-   [x] Any dependent changes have been merged and published in downstream modules
-   [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -89,6 +89,7 @@
     "interval-plus": "^1.0.2",
     "lodash.debounce": "^4.0.8",
     "mixpanel-browser": "^2.46.0",
+    "moment": "^2.29.4",
```
Comment (shilomagen) [inline package.json:92]: We are not using `moment` lib as the date utils library, we use `date-fns`. 
Can you please use `date-fns`? Thanks!

File: src/components/App/App.test.tsx [hunk 2]
```diff
@@ -1,12 +1,13 @@
 import React from 'react';
-import { render, waitFor } from '@testing-library/react';
-import Content from '@src/content.json';
 import { App } from './App';
+import { PageBaseDriver } from '@test/RTL/RTLPageBaseDriver';
 
 describe('App', () => {
+  const driver = new PageBaseDriver();
```
Comment (shilomagen) [inline src/components/App/App.test.tsx:6]: Beautiful

File: src/components/DateRangePicker/DateRangePicker.test.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import moment from 'moment/moment';
+import { IsraelDateDigitsFormat } from '@src/lib/utils';
+import { PageBaseDriver } from '@test/RTL/RTLPageBaseDriver';
+
+describe('Date Range Picker', () => {
+  const driver = new PageBaseDriver();
+  const yesterdayDate = moment().subtract(1, 'days').format(IsraelDateDigitsFormat);
```
Comment (shilomagen) [inline src/components/DateRangePicker/DateRangePicker.test.tsx:8]:
```
import subDays from 'date-fns/subDays'
const yesterday = subDays(new Date(), 1)
```

**Review by shilomagen — CHANGES_REQUESTED** [review]
Amazing job, let's switch to date-fns and I will merge!
Thanks @yair-dana !

**Comment by yair-dana** [comment]
Hi @shilomagen ,
I made the changes following the review.
_I also handle some edge case related to storage:_
when you set a start/end date and after a few days those dates are not valid (dates that already pass).
in this case, we should generate the default dates.

WDYT following my open question?
```
There is a decision to be made on how to implement this feature - either by calling the MyVisit service with the updated Start Date, or by filtering the dates after they have been fetched from the service. (For now- both approaches are implemented)
My concern is that the startDate functionality may not be properly implemented on the MyVisit side.
```
Thanks!

**Review by shilomagen — APPROVED** [review]

**Pull request merged**